torch/lib/c10d/reducer.hpp

```diff
@@ -96,7 +118,8 @@ class Reducer {
   // buckets once after the first iteration and never rebuild them if
   // find_unused_parameters_.
   inline bool should_rebuild_buckets() const {
-    return !find_unused_parameters_ && !has_rebuilt_bucket_;
+    return !find_unused_parameters_ && !has_rebuilt_bucket_
+      && (unused_parameters_.size() == 0);
```

Comment:
Why do we need this additional check for `unused_parameters`? Is there a scenario where `unused_parameters` will be non-zero despite `find_unused_parameters=True`?

Reply:
yes, if users set find_unused_parameters=false and only set_static_graph(), it is possible unused_parameters_ size is non zero